tools-v2/pkg/cli/command/curvebs/check/chunkserver/chunkserver_info.go

```diff
@@ -0,0 +1,188 @@
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+ */
+/*
+* Project: CurveCli
+* Created Date: 2023-06-02
+* Author: montaguelhz
+ */
+package chunkserver
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	cmderror "github.com/opencurve/curve/tools-v2/internal/error"
+	basecmd "github.com/opencurve/curve/tools-v2/pkg/cli/command"
+	"github.com/opencurve/curve/tools-v2/pkg/config"
+	"github.com/opencurve/curve/tools-v2/pkg/output"
+	"github.com/opencurve/curve/tools-v2/proto/proto/topology"
+	"github.com/opencurve/curve/tools-v2/proto/proto/topology/statuscode"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"google.golang.org/grpc"
+)
+
+type GetChunkServerInfoRpc struct {
+	Info      *basecmd.Rpc
+	Request   *topology.GetChunkServerInfoRequest
+	mdsClient topology.TopologyServiceClient
+}
+
+var _ basecmd.RpcFunc = (*GetChunkServerInfoRpc)(nil)
+
+func (gRpc *GetChunkServerInfoRpc) NewRpcClient(cc grpc.ClientConnInterface) {
+	gRpc.mdsClient = topology.NewTopologyServiceClient(cc)
+}
+
+func (gRpc *GetChunkServerInfoRpc) Stub_Func(ctx context.Context) (interface{}, error) {
+	return gRpc.mdsClient.GetChunkServer(ctx, gRpc.Request)
+}
+
+type ChunkServerInfoCommand struct {
+	basecmd.FinalCurveCmd
+	Rpc   []*GetChunkServerInfoRpc
+	infos []*topology.ChunkServerInfo
+}
+
+var _ basecmd.FinalCurveCmdFunc = (*ChunkServerInfoCommand)(nil)
+
+func (csiCmd *ChunkServerInfoCommand) AddFlags() {
+	config.AddBsMdsFlagOption(csiCmd.Cmd)
+	config.AddRpcRetryTimesFlag(csiCmd.Cmd)
+	config.AddRpcTimeoutFlag(csiCmd.Cmd)
+
+	config.AddBsChunkServerIdOptionFlag(csiCmd.Cmd)
+	config.AddBsChunkServerAddrOptionFlag(csiCmd.Cmd)
+}
+
+func NewGetChunkServerInfoCommand() *ChunkServerInfoCommand {
+	csiCmd := &ChunkServerInfoCommand{FinalCurveCmd: basecmd.FinalCurveCmd{}}
+	basecmd.NewFinalCurveCli(&csiCmd.FinalCurveCmd, csiCmd)
+	return csiCmd
+}
+
+func NewChunkServerInfoCommand() *cobra.Command {
+	return NewGetChunkServerInfoCommand().Cmd
+}
+
+func (csiCmd *ChunkServerInfoCommand) Init(cmd *cobra.Command, args []string) error {
+	mdsAddrs, err := config.GetBsMdsAddrSlice(csiCmd.Cmd)
+	if err.TypeCode() != cmderror.CODE_SUCCESS {
+		return err.ToError()
+	}
+	timeout := config.GetFlagDuration(csiCmd.Cmd, config.RPCTIMEOUT)
+	retrytimes := config.GetFlagInt32(csiCmd.Cmd, config.RPCRETRYTIMES)
+
+	var ids []uint32
+	var addrs []string
+	if viper.IsSet(config.VIPER_CURVEBS_CHUNKSERVER_ADDR) && !viper.IsSet(config.VIPER_CURVEBS_CHUNKSERVER_ID) {
+		// chunkserveraddr is set, but chunkserverid is not set
+		addrs, err = config.GetBsChunkserverAddrSlice(csiCmd.Cmd)
+		if err.TypeCode() != cmderror.CODE_SUCCESS {
+			return fmt.Errorf(err.Message)
+		}
+	} else {
```

Comment:
else contains 3 cases:
1. `viper.IsSet(config.VIPER_CURVEBS_CHUNKSERVER_ADDR) && viper.IsSet(config.VIPER_CURVEBS_CHUNKSERVER_ID)`
2. `!viper.IsSet(config.VIPER_CURVEBS_CHUNKSERVER_ADDR) && !viper.IsSet(config.VIPER_CURVEBS_CHUNKSERVER_ID)`
3. `!viper.IsSet(config.VIPER_CURVEBS_CHUNKSERVER_ADDR) && viper.IsSet(config.VIPER_CURVEBS_CHUNKSERVER_ID)`
Is that your logic?